lib/material/material_gui.flow

```diff
@@ -1468,4 +1472,200 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
 		\sz -> MGetFocusGroup(\parent -> TFrameT(margin, radius, fselectLift(style, \s -> mgraphicsStyle2tgraphicsStyle(parent, s)), sz)),
 		true
 	)
+}
+
+MSegmentedButtons(
+	items : [Pair<Pair<Maybe<MText>, Maybe<MIcon>>, Pair<[MButtonState], DynamicBehaviour<bool>>>],
+	multiSelect : bool,
+	iconsOnly : bool,
+	size : WidthHeight,
+	graphicsStyles : [MGraphicsStyle]
+) -> Material {
+	defaultBgColor = extractStruct(graphicsStyles, MThemedFill(MSecondaryColor())).color;
+	defaultStrokeColor = extractStruct(graphicsStyles, MThemedStroke(MSecondaryColor())).color;
+
+    minCellWidthIcons = 40.0;
+    minCellWidthText = 60.0;
+    charWidthMultiplier = 8.0;
+    textPadding = 32.0;
+    iconPadding = 24.0;
+    cellPadding = 16.0;
+    buttonContentMargin = 48.0;
+
+	totalWidth = size.width;
+	totalHeight = size.height;
+	borderRadius = totalHeight / 2.0; // 50%
+
+	// Calculate required width for each button based on content
+	requiredWidths = map(items, \item -> {
+		title = if (iconsOnly) "" else eitherFn(item.first.first, \text -> text.text, \ -> "");
+		hasIcon = isSome(item.first.second);
+		textWidth = i2d(strlen(title)) * charWidthMultiplier;
+		padding = if (iconsOnly) iconPadding else textPadding;
+
+		if (iconsOnly) minCellWidthIcons + padding
+		else textWidth + minCellWidthIcons + padding;
+	});
+
+	// Optimize cell width distribution
+	calculateCellWidths = {
+		availableWidth = totalWidth;
+		itemCount = length(items);
+		minCellWidth = if (iconsOnly) minCellWidthIcons else minCellWidthText;
+
+		needsMore = map(requiredWidths, \w -> w > (availableWidth / i2d(itemCount)));
+		withEnoughSpace = mapi(requiredWidths, \i, w -> if (!needsMore[i]) w else 0.0);
+		needMoreSpace = mapi(requiredWidths, \i, w -> if (needsMore[i]) w else 0.0);
+
+		spaceUsedByEnough = fold(withEnoughSpace, 0.0, \acc, w -> acc + w);
+		needMoreCount = fold(needsMore, 0, \acc, need -> if (need) acc + 1 else acc);
+		enoughSpaceCount = itemCount - needMoreCount;
+        
+		if (needMoreCount == 0) {
+			remainingSpace = availableWidth - spaceUsedByEnough;
+			extraPerCell = remainingSpace / i2d(itemCount);
+			map(requiredWidths, \w -> max(w + extraPerCell, minCellWidth));
+		} else {
+			remainingSpace = availableWidth - spaceUsedByEnough;
+			equalShare = remainingSpace / i2d(needMoreCount);
+			mapi(requiredWidths, \i, w ->
+				if (needsMore[i]) max(equalShare, minCellWidth)
+				else max(w, minCellWidth)
+			);
+		}
+	}
+
+	cellWidths = calculateCellWidths;
+	lastSelected = make(None());
+	buttonsFocus = generate(0, length(items), \__ -> make(false));
+
+	makeButton = \i, item, isInteractive -> {
```

Comment:
I'd call makeButton once and return Pair<Material, Material>. WIth this impkementation you don't need to create separate buttonsFocus array.